File: shoprbackend/src/main/java/com/realdolmen/shoprbackend/domain/Lp.java
package com.realdolmen.shoprbackend.domain;

import javax.persistence.*;
import javax.validation.constraints.Size;

@Entity
@Table(name="lps")
@DiscriminatorValue("lp")
public class Lp extends Item {


    @Size(max=100)
    private String publisher;

    @Enumerated(EnumType.STRING)
    private LpGenre genre;


    public String getPublisher() {
        return publisher;
    }

    public void setPublisher(String publisher) {
        this.publisher = publisher;
    }

    public LpGenre getGenre() {
        return genre;
    }

    public void setGenre(LpGenre genre) {
        this.genre = genre;
    }
}

File: frontend/src/app/models/lp.ts
import {Article} from "./article";
import {LpGenre} from "./lp-genre";

export class Lp extends Article {
  public publisher: string;
  public lpGenre: LpGenre;


  constructor(publisher?: string, lpGenre?: LpGenre) {
    super();
    this.publisher = publisher;
    this.lpGenre = lpGenre;
  }
}
